src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -1587,7 +1587,8 @@ public RubyNode visitGlobalAsgnNode(GlobalAsgnParseNode node) {
         final RubyNode translatedValue = node.getValueNode().accept(this);
         final String variableName = node.getName();
 
-        return new LazyRubyNode(() -> writeGlobal(node, variableName, translatedValue));
+        return new LazyRubyNode(() -> node.getPosition().toSourceSection(source),
+                () -> writeGlobal(node, variableName, translatedValue));
```

Comment:
efa975ebd6e82d613d8bf54d09d35d7564a723ec added those lazy nodes for globals.
I think it is needed so we can lookup if there is an alias at the first execution and not too early during parse time.